### PR TITLE
Add patterns

### PR DESCRIPTION
**Pull request by km1414**
**ruleIntervalDOMDOM** - now only extracts dates in current month, if-else condition didn't work for switching between current and next month.

### CODE REVIEW

File: Duckling/Time/EN/Rules.hs [hunk 1]
```diff
@@ -123,6 +123,24 @@ ruleYYYYNamedMonthDOM = Rule
...
   }
 
+ruleNamedMonthDOMYYYY :: Rule
+ruleNamedMonthDOMYYYY = Rule
+  { name = "<named-month> <day-of-month> <year>"
+  , pattern =
+    [ Predicate isAMonth
+    , regex "\\s*?\\-\\s*?|\\s*?/\\s*?|\\s*?_+\\s*?|\\s*?and\\s*?|\\s*?to\\s*?|\\s*?th?ru\\s*?|\\s*?through\\s*?|\\s*?(un)?til(l)?\\s*?|\\s+"
+    , Predicate isDOMValue
+    , regex "\\s*?\\-\\s*?|\\s*?/\\s*?|\\s*?_+\\s*?|\\s*?and\\s*?|\\s*?to\\s*?|\\s*?th?ru\\s*?|\\s*?through\\s*?|\\s*?(un)?til(l)?\\s*?|\\s+"
```
Comment (pranas) [inline Duckling/Time/EN/Rules.hs]: I think you can make this more readable by extracting the common `\\s*`, it should be enough to have it at the beginning and the end

Comment (km1414) [reply Duckling/Time/EN/Rules.hs]: Cool, seems like fixed that.

File: Duckling/Time/EN/Rules.hs [hunk 2]
```diff
@@ -1148,14 +1184,35 @@ ruleIntervalDDDDMonth = Rule
...
+
 ruleIntervalFromMonthDDDD :: Rule
 ruleIntervalFromMonthDDDD = Rule
   { name = "from <month> dd-dd (interval)"
   , pattern =
     [ regex "from|between"
     , Predicate isAMonth
     , Predicate isDOMValue
-    , regex "\\-|and|to|th?ru|through|(un)?til(l)?"
+    , regex "\\s*?(\\-|/|_+|and|to|th?ru|through|(un)?til(l)?)\\s*?|\\s+"
```
Comment (pranas) [inline Duckling/Time/EN/Rules.hs:1215]: is the last `\\s+` correct?

Comment (pranas) [reply Duckling/Time/EN/Rules.hs:1215]: nevermind, got it

**Review by pranas — APPROVED** [review]

**Pull request merged**